### PR TITLE
added test for refund expiry and ability to view expiry time

### PR DESCRIPTION
**Pull request by royosherove**
Hey folks. As discussed in the twitter space, here's an implementation of the function to view current expiry time, plus a test for the refund reverting after expiry time.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "workbench.colorCustomizations": {
+        "activityBar.background": "#213030",
+        "titleBar.activeBackground": "#2E4444",
+        "titleBar.activeForeground": "#F9FBFB"
+    }
+}
```
Comment (elie222) [inline .vscode/settings.json:7]: can we add this to `.gitignore` so not everyone has to accept these settings, please?

File: contracts/ERC721RExample.sol [hunk 2]
```diff
@@ -94,10 +95,14 @@ contract ERC721RExample is ERC721A, Ownable {
         Address.sendValue(payable(msg.sender), refundAmount);
     }
 
-    function refundGuaranteeActive() public view returns (bool) {
+    function getRefundGuaranteeEndTime() public view returns (uint256) {
```
Comment (elie222) [inline contracts/ERC721RExample.sol:98]: `public refundEndTime` doesn't have the same effect?

File: contracts/ERC721RExample.sol [hunk 3]
```diff
@@ -122,6 +127,8 @@ contract ERC721RExample is ERC721A, Ownable {
 
     function toggleRefundCountdown() public onlyOwner {
         refundEndTime = block.timestamp + refundPeriod;
+        console.log("Refund counted started to block");
```
Comment (elie222) [inline contracts/ERC721RExample.sol:130]: if we can remove logs from the pr please :)

File: contracts/cryptofighters/CryptoFightersV2.sol [hunk 4]
```diff
@@ -130,7 +130,10 @@ contract CryptoFightersV2 is ERC721A, Ownable, ReentrancyGuard {
         Address.sendValue(payable(msg.sender), refundAmount);
     }
 
-    function refundGuaranteeActive() public view returns (bool) {
+    function getRefundGuaranteeEndBlock() public view returns (uint256) {
```
Comment (elie222) [inline contracts/cryptofighters/CryptoFightersV2.sol:133]: `public refundEndTime` instead?

**Review by elie222 — CHANGES_REQUESTED** [review]
thanks for the pr! made some small comments before merging in :)

File: .vscode/settings.json (hunk 1 above)
Comment (royosherove) [reply .vscode/settings.json:7]: yes - that one slipped through!

File: contracts/ERC721RExample.sol (hunk 2 above)
Comment (royosherove) [reply contracts/ERC721RExample.sol:98]: a function can be a more explicit way to declare this specifically as part of a standard. a field is more implicit and deals with implementation details. Ideally you want the interface to not care about implementation.

File: contracts/ERC721RExample.sol (hunk 3 above)
Comment (royosherove) [reply contracts/ERC721RExample.sol:130]: Yup

File: contracts/cryptofighters/CryptoFightersV2.sol (hunk 4 above)
Comment (royosherove) [reply contracts/cryptofighters/CryptoFightersV2.sol:133]: More of a style choice. I feel get makes it clear it is a read from the caller POV.

Comment (royosherove) [reply contracts/cryptofighters/CryptoFightersV2.sol]: oh sorry I misunderstood. function has been renamed

**Comment by royosherove** [comment]
went through all requested changes and made them except the "get" thing. take a look at the updated files

File: contracts/ERC721RExample.sol (hunk 2 above)
Comment (elie222) [reply contracts/ERC721RExample.sol:97]: so probably what we want to do is extact this code into its own contract, and it have it implement `IERC721R`. and you're right. seems like even with ERC721 it's this way with getApproved

**Comment by elie222** [comment]
thanks! :)

**Pull request merged**